osu.Game/Overlays/Comments/CommentsContainer.cs

```diff
@@ -13,37 +12,82 @@
 using System.Threading;
 using System.Linq;
 using osu.Framework.Extensions.IEnumerableExtensions;
+using osu.Game.Users;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Graphics.Sprites;
+using osu.Framework.Threading;
+using System;
 
 namespace osu.Game.Overlays.Comments
 {
-    public class CommentsContainer : CompositeDrawable
+    public abstract class CommentsContainer : CompositeDrawable
```

Comment:
making this abstract *just* for testing feels very very backwards.

Reply:
So, there's no reason to make a separate `OnlineCommentsContainer` at all or I should make the APIRequest virtual with null in a base one?

Reply:
the default and only usage in osu! is "online". so the implementation should be as such. class inheritance shouldn't be structured around making tests work – just do this via virtual `Create` or settable properties.